From 6f605598de1eb4361ef912e85edb27a8b7b96b21 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 13 Jul 2012 00:01:51 +0000
Subject: global: don't run a garbage collection on tweeners end

This currently causes the shell to freeze very often in a thread
deadlock, and the gjs garbage collector behavior is currently getting
fixed at the right level in gjs itself.

https://bugzilla.gnome.org/show_bug.cgi?id=679832
---
diff --git a/src/shell-global.c b/src/shell-global.c
index bb32a01..28127ab 100644
--- a/src/shell-global.c
+++ b/src/shell-global.c
@@ -1482,13 +1482,6 @@ run_leisure_functions (gpointer data)
   if (global->work_count > 0)
     return FALSE;
 
-  /* Previously we called gjs_maybe_gc().  However, it simply doesn't
-   * trigger often enough.  Garbage collection is very fast here, so
-   * let's just aggressively GC.  This will help avoid both heap
-   * fragmentation, and the GC kicking in when we don't want it to.
-   */
-  gjs_context_gc (global->js_context);
-
   /* No leisure closures, so we are done */
   if (global->leisure_closures == NULL)
     return FALSE;
--
cgit v0.9.0.2
